EIPS/eip-1153.md

```diff
@@ -1,36 +1,48 @@
 created: 2018-06-15
 ---
 
-## Simple Summary
-Support for efficient transient storage in EVM. It is like regular storage (`SLOAD`/`SSTORE`), but with the lifetime limited to one Ethereum transaction.
-Notable use case is efficient reentrancy lock.
-
 ## Abstract
-This proposal introduces transient storage, which behaves similar to storage,
-but the updates will only persist within one Ethereum transaction. Transient storage is accessible to smart contracts via new opcodes: `TLOAD` and `TSTORE` (“T” stands for Transient).
+This proposal introduces transient storage opcodes, which manipulate state that behaves identically to storage, 
+but is discarded after every transaction. Transient storage is cheaper since it never requires disk access.
+Transient storage is accessible to smart contracts via 2 new opcodes: `TLOAD` and `TSTORE`, where “T” stands for "transient."
 
 ## Motivation
 Running a transaction in Ethereum can generate multiple nested frames of execution, each created by `CALL` (or similar) instructions.
 Contracts can be re-entered during the same transaction, in which case there are more than one frame belonging to one contract.
 Currently, these frames can communicate in two ways - via inputs/outputs passed via `CALL` instructions, and via storage updates.
-If there is an intermediate frame belonging to another contract, communication via inputs/outputs is not secure. Notable example is a reentrancy lock which cannot rely on the intermediate frame to pass through the state of the lock.
+If there is an intermediate frame belonging to another untrusted contract, communication via inputs/outputs is not secure. 
+Notable example is a reentrancy lock which cannot rely on the intermediate frame to pass through the state of the lock.
 Communication via storage (`SSTORE`/`SLOAD`) is costly. Transient storage is a dedicated and gas efficient solution to the problem of inter frame communication.
 
-Language support could be added in relatively easy way. For example, in Solidity, a qualifier “transient” can be introduced (similar to the existing qualifiers “memory” and “storage”). Since addressing scheme of `TSTORE` and `TLOAD` is the same as for `SSTORE` and `SLOAD`, code generation routines that exist for storage variables, can be easily generalised to also support transient storage.
+Storage refunds accumulated due to inter frame communication are also limited to 20% of gas spent by a transaction as of [EIP-3529](./eip-3529.md) included in the London hard fork. 
+This greatly reduces the refunds for transiently-set storage slots in otherwise low-cost transactions. For example, in order to receive the full refund of one re-entrancy lock, 
+the transaction must spend ~80k gas on other operations.
+
+Language support could be added in relatively easy way. For example, in Solidity, a qualifier “transient” can be introduced 
+(similar to the existing qualifiers “memory” and “storage”, and Java's own `transient` keyword with a similar meaning). 
```

Comment:
I would not bring Java here, as far as I remember `transient` in java simply means that a class member does not participate in the serialisation when built-in serialisation library is used. But I kind of see the similarity in the meaning :)

Reply:
that's right yeah, but often in java a class would be serialized for writing to storage, so it did feel relevant in making the keyword more digestible